plugins/out_s3/s3.c

```diff
@@ -101,6 +115,50 @@ struct flb_aws_header *create_canned_acl_header(char *canned_acl)
     return acl_header;
 };
 
+static struct flb_aws_header *create_headers(struct flb_s3 *ctx, int num_headers)
+{
+    int n = 0;
+    struct flb_aws_header *canned_acl_header = NULL;
+    struct flb_aws_header *headers = NULL;
+    if (num_headers == 0) {
+        return NULL;
+    }
+
+    headers = flb_malloc(sizeof(struct flb_aws_header) * num_headers);
+    if (headers == NULL) {
+        flb_errno();
+        return NULL;
+    }
+
+    if (ctx->content_type != NULL) {
+        content_type_header.val = ctx->content_type;
+        content_type_header.val_len = strlen(ctx->content_type);
+        headers[n] = content_type_header;
```

Comment:
Since this plugin can make concurrent requests... I am pretty sure this is not safe. Multiple requests could use the header declared with `static` at the same time.

Your headers array is heap allocated memory. Instead of this:

```
 headers[n] = content_type_header;
```

Use memcpy to copy the static header into the array of memory and then modify it